python/cuml/dask/datasets/make_regression.py

```diff
@@ -0,0 +1,216 @@
+    """Generate a random regression problem.
+    The input set can either be well conditioned (by default) or have a low
+    rank-fat tail singular profile.
+
+    The output is generated by applying a (potentially biased) random linear
+    regression model with "n_informative" nonzero regressors to the previously
+    generated input and some gaussian centered noise with some adjustable
+    scale.
+
+    Parameters
+    ----------
+    n_samples : int, optional (default=100)
+        The number of samples.
+    n_features : int, optional (default=100)
+        The number of features.
+    n_informative : int, optional (default=10)
+        The number of informative features, i.e., the number of features used
+        to build the linear model used to generate the output.
+    n_targets : int, optional (default=1)
+        The number of regression targets, i.e., the dimension of the y output
+        vector associated with a sample. By default, the output is a scalar.
+    bias : float, optional (default=0.0)
+        The bias term in the underlying linear model.
+    effective_rank : int or None, optional (default=None)
+        if not None:
+            The approximate number of singular vectors required to explain most
+            of the input data by linear combinations. Using this kind of
+            singular spectrum in the input allows the generator to reproduce
+            the correlations often observed in practice.
+        if None:
+            The input set is well conditioned, centered and gaussian with
+            unit variance.
+    tail_strength : float between 0.0 and 1.0, optional (default=0.5)
+        The relative importance of the fat noisy tail of the singular values
+        profile if "effective_rank" is not None.
+    noise : float, optional (default=0.0)
+        The standard deviation of the gaussian noise applied to the output.
+    shuffle : boolean, optional (default=False)
+        Shuffle the samples and the features.
+    coef : boolean, optional (default=False)
+        If True, the coefficients of the underlying linear model are returned.
+    random_state : int, RandomState instance or None (default)
+        Determines random number generation for dataset creation. Pass an int
+        for reproducible output across multiple function calls.
+    n_parts : int, optional (default=1)
+        The number of parts of work.
+
+    Returns
+    -------
+    X : Dask-cuPy array of shape [n_samples, n_features]
+        The input samples.
+    y : Dask-cuPy array of shape [n_samples] or [n_samples, n_targets]
+        The output values.
+    coef : Dask-cuPy array of shape [n_features]
+           or [n_features, n_targets], optional
+        The coefficient of the underlying linear model. It is returned only if
+        coef is True.
+    """
+    n_informative = min(n_features, n_informative)
+    rs = da.random.RandomState(seed=random_state,
+                               RandomState=cp.random.RandomState)
```

Comment:
This is great that we are able to initialize the Dask.Array random state w/ the CuPy random state. We should do this in the above function as well.